RFC/src/RFC-0201_TariScript.md

```diff
@@ -269,164 +264,162 @@ pub struct TransactionInput {
 ```
 
 In standard Mimblewimble, an input is the same as an output _sans_ range proof. The range proof doesn't need to be checked
-again when spending inputs, so it is dropped. However, under this scheme, the range proof must be checked again to ensure
-that the output metadata (script, output features) have not been modified.
+again when spending inputs, so it is dropped. 
+
 
-_Note:_ Currently, the output features are actually malleable. TariScript fixes this.
 
 ```rust,ignore
 pub struct TransactionInput {
     /// Options for an output's structure or use
     features: OutputFeatures,
     /// The homomorphic commitment representing the output amount
     commitment: Commitment,
-    /// A proof that the commitment is in the right range
-    proof: RangeProof,
     /// The serialised script
     script: Vec<u8>,
-    /// The public script key, L
-    script_pub_key: PublicKey
     /// The script input data, if any
     input_data: Vec<u8>
-    /// A signature with k', signing the script and input data
-    script_signature: Signature
+    /// A signature with K_b, signing the script and input data
+    script_signature: Signature,
```

Comment:
How does this differ from the owner_offset which already  proves knowledge of the receiver private key?

Reply:
Signs the script with the owner offset key. When locks the script and the script input data to not be malleable. The script might have extra spending constrictions on it, that is not in the normal owner offset.